assets/blocks/course-outline/course-block/edit.js

```diff
@@ -35,10 +34,48 @@ const useSynchronizeLessonsOnUpdate = function ( clientId ) {
 	);
 
 	useEffect( () => {
-		dispatch( COURSE_STATUS_STORE ).stopTrackingRemovedLessons(
-			outlineDescendants
-		);
-	}, [ clientId, outlineDescendants ] );
+		if ( ! isPreview ) {
+			dispatch( COURSE_STATUS_STORE ).stopTrackingRemovedLessons(
+				outlineDescendants
+			);
+		}
+	}, [ clientId, outlineDescendants, isPreview ] );
+};
+
+const useApplyStyleToModules = ( clientId, className, isPreview ) => {
+	const [ oldOutlineClass, setOutlineClass ] = useState( null );
+	const outlineStyles = useSelect(
+		( select ) =>
+			select( 'core/blocks' ).getBlockStyles(
+				'sensei-lms/course-outline'
+			),
+		[]
+	);
+
+	const newOutlineClass = getActiveStyleClass( outlineStyles, className );
+
+	// setOutlineClass is called when there is an update in course style class only.
+	// eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Similar to the other comment, it would have basically the same result, but we could add the dependencies to follow the good practices. So we run this effect only when the dependency variables were changed. :)

Reply:
Argh I think this is a leftover from a previous solution which required the effect to run on every render. Added dependencies and switched to `useRef` in a4e76b0ce44194a5c8d3e77f92767fb6860ef15d